From bd94ba0f6020a8ac59d6714f84e0f0c8538b902f Mon Sep 17 00:00:00 2001
From: Evan Wang <xswang@marvell.com>
Date: Wed, 5 Jul 2017 14:11:13 +0800
Subject: [PATCH 1775/2241] fix: pci: aardvark: reconfig max payload size when
 resume

Currently the Max Payload Size is not restored when
S2R resume, which causes PCIE data transfer problem
after resume.
The patch restored the max payload size in resume
routine.
JIRA number: A3700-1357

Change-Id: If0a3f12d0c26dbb13f5018e5da96ee3dbacd8bb6
Signed-off-by: Evan Wang <xswang@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/41159
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
---
 drivers/pci/host/pci-aardvark.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/host/pci-aardvark.c b/drivers/pci/host/pci-aardvark.c
index 265aa89..147f9b0 100644
--- a/drivers/pci/host/pci-aardvark.c
+++ b/drivers/pci/host/pci-aardvark.c
@@ -177,6 +177,7 @@
 
 struct advk_pcie {
 	struct platform_device *pdev;
+	struct pci_bus *bus;
 	void __iomem *base;
 	struct phy *phy;
 	struct list_head resources;
@@ -904,9 +905,10 @@ static int advk_pcie_bus_configure_mps(struct pci_dev *dev, void *data)
 	return 0;
 }
 
-static void advk_pcie_configure_mps(struct pci_bus *bus, struct advk_pcie *pcie)
+static void advk_pcie_configure_mps(struct advk_pcie *pcie)
 {
 	u8 smpss = PCIE_CORE_DEV_CTRL_STATS_MAX_PAYLOAD_SZ;
+	struct pci_bus *bus = pcie->bus;
 	u32 reg;
 
 	/* Find the minimal supported MAX payload size */
@@ -1060,12 +1062,13 @@ after_pcie_reset:
 	}
 
 	pci_bus_assign_resources(bus);
+	pcie->bus = bus;
 
 	list_for_each_entry(child, &bus->children, node)
 		pcie_bus_configure_settings(child);
 
 	/* Configure the MAX pay load size */
-	advk_pcie_configure_mps(bus, pcie);
+	advk_pcie_configure_mps(pcie);
 
 	pci_bus_add_devices(bus);
 
@@ -1122,6 +1125,9 @@ static int advk_pcie_resume_noirq(struct device *dev)
 
 	advk_pcie_setup_hw(pcie);
 
+	/* Reconfigure the MAX pay load size */
+	advk_pcie_configure_mps(pcie);
+
 	return 0;
 }
 
-- 
2.7.4

